### DISCUSSION TITLE
How to customize Lazy dashboard?

### DISCUSSION THREAD

**Discussion by tonmoy998** [LazyVim/LazyVim#742] [Q&A] [answered]
I want to change lazy banner.

**Comment by dpetka2001** [comment, marked as answer]
Something like this should work
```lua
  {
    "goolord/alpha-nvim",
    opts = function(_, dashboard)
      local logo = [[ test ]]
      dashboard.section.header.val = vim.split(logo, "\n")
    end,
  },
```
Just change the `logo` variable to whatever you want.

**Reply by tonmoy998** [reply]
Thank you so much 💙

**Comment by linsui** [comment]
What should I do if I want to add/remove buttons?

**Reply by dpetka2001** [reply]
Haven't done this but you should take a look at the [docs](https://www.lazyvim.org/plugins/extras/util.project#alpha-nvim) to get a grasp of how it can be achieved. Basically you should define the button to add and add it with `table.insert` and you could use `table.remove` to remove buttons.

**Reply by linsui** [reply]
Wow, thanks! I missed that part...

**Comment by Zacktamondo** [comment]
Since LazyVim switched to dashboard-nvim, how can I change  the logo and the theme now?

**Reply by Zacktamondo** [reply]
Found it:
```
return {
  {
    "nvimdev/dashboard-nvim",
    event = "VimEnter",
    opts = function(_, opts)
      local logo = [[
LOGO
            ]]

      logo = string.rep("\n", 8) .. logo .. "\n\n"
      opts.config.header = vim.split(logo, "\n")
      opts.theme = "doom"
    end,
  },
}
```

**Reply by anonbread** [reply]
u saved my life bro thanks a lot :]

**Reply by MDSADABWASIM** [reply]
I've this in `config/plugins/dashboard.lua` but it's not working:

```

return {
  {
    "nvimdev/dashboard-nvim",
    event = "VimEnter",
    opts = function(_, opts)
      local logo = [[
                                                                               
                                                                LOGO                                   
                                                                               
    ]]

      logo = string.rep("\n", 8) .. logo .. "\n\n"
      opts.config.header = vim.split(logo, "\n")
      opts.theme = "doom"
    end,
  },
}

```

Am I missing something @Zacktamondo ?

**Reply by dotfrag** [reply]
The code you provided works, you might have it in the wrong path. It should be `lua/plugins/dashboard.lua`.

Notice `config` -> `lua`.

**Reply by Babbili** [reply]
@Zacktamondo this exact code in `.config/nvim/lua/plugins/dashboard.lua` i'm getting an error 
<img width="669" height="231" alt="image" src="https://github.com/user-attachments/assets/c1240db2-14d4-4ad3-b81b-2c112192f5e8" />

**Reply by HamzaHeikal2011** [reply]
i have the same issue as @Babbili

**Reply by hzionn** [reply]
@Babbili @HamzaHeikal2011 my current setup works fine for me

```
-- .config/nvim/lua/plugins/dashboard.lua
return {
  "snacks.nvim",
  opts = {
    dashboard = {
      -- dashboard configuration
      preset = {
        keys = {
          { icon = " ", key = "f", desc = "Find File", action = ":lua Snacks.dashboard.pick('files')" },
          -- { icon = " ", key = "n", desc = "New File", action = ":ene | startinsert" },
          -- { icon = " ", key = "g", desc = "Find Text", action = ":lua Snacks.dashboard.pick('live_grep')" },
          { icon = " ", key = "r", desc = "Recent Files", action = ":lua Snacks.dashboard.pick('oldfiles')" },
          {
            icon = " ",
            key = "c",
            desc = "Config",
            action = ":lua Snacks.dashboard.pick('files', {cwd = vim.fn.stdpath('config')})",
          },
          -- { icon = " ", key = "s", desc = "Restore Session", section = "session" },
          -- { icon = "󰒲 ", key = "L", desc = "Lazy", action = ":Lazy", enabled = package.loaded.lazy ~= nil },
          { icon = " ", key = "q", desc = "Quit", action = ":qa" },
        },
        header = [[
⠀⠀⠀⠀⠀⢸⠓⢄⡀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⢸⠀⠀⠑⢤⡀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⢸⡆⠀⠀⠀⠙⢤⡷⣤⣦⣀⠤⠖⠚⡿⠁⠀⠀⠀⠀⠀⠀⠀⠀⠀
⣠⡿⠢⢄⡀⠀⡇⠀⠀⠀⠀⠀⠉⠀⠀⠀⠀⠀⠸⠷⣶⠂⠀⠀⠀⣀⣀⠀⠀⠀
⢸⣃⠀⠀⠉⠳⣷⠞⠁⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠈⠉⠉⠉⠉⠉⠉⠉⢉⡭⠋
⠀⠘⣆⠀⠀⠀⠁⠀⢀⡄⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢀⡴⠋⠀⠀
⠀⠀⠘⣦⠆⠀⠀⢀⡎⢹⡀⠀⠀⠀⠀⠀⠀⠀⠀⡀⠀⠀⡀⣠⠔⠋⠀⠀⠀⠀
⠀⠀⠀⡏⠀⠀⣆⠘⣄⠸⢧⠀⠀⠀⠀⢀⣠⠖⢻⠀⠀⠀⣿⢥⣄⣀⣀⣀⠀⠀
⠀⠀⢸⠁⠀⠀⡏⢣⣌⠙⠚⠀⠀⠠⣖⡛⠀⣠⠏⠀⠀⠀⠇⠀⠀⠀⠀⢙⣣⠄
⠀⠀⢸⡀⠀⠀⠳⡞⠈⢻⠶⠤⣄⣀⣈⣉⣉⣡⡔⠀⠀⢀⠀⠀⣀⡤⠖⠚⠀⠀
⠀⠀⡼⣇⠀⠀⠀⠙⠦⣞⡀⠀⢀⡏⠀⢸⣣⠞⠀⠀⠀⡼⠚⠋⠁⠀⠀⠀⠀⠀
⠀⢰⡇⠙⠀⠀⠀⠀⠀⠀⠉⠙⠚⠒⠚⠉⠀⠀⠀⠀⡼⠁⠀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⢧⡀⠀⢠⡀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠙⣞⠁⠀⠀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⠙⣶⣶⣿⠢⣄⡀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢸⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠉⠀⠀⠀⠙⢿⣳⠞⠳⡄⠀⠀⠀⢀⡞⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠉⠀⠀⠹⣄⣀⡤⠋⠀⠀⠀⠀⠀hzionn
        ]],
      },
    },
  },
}
```

**Reply by HamzaHeikal2011** [reply]
Got there after some research. Tysm tho

**Comment by dotfrag** [comment]
You can have a look at my config for inspiration https://github.com/dotfrag/nvim/blob/main/lua/plugins/dashboard.lua

Tagging @MDSADABWASIM for visibility.

**Reply by MDSADABWASIM** [reply]
Hey,

Thanks for the reply,

I tried running your dashboard config file but it didn't work.
If you want to check this is [my config file.](https://github.com/MDSADABWASIM/.dotfiles/blob/master/nvim/lua/plugins/dashboard.lua)

**Reply by dotfrag** [reply]
You have alpha dashboard enabled in your lazy extras (lazyvim.json file) and it's disabling the nvimdev dashboard.

**Reply by MDSADABWASIM** [reply]
Thanks it works after removing that.

**Reply by SEBASCMB** [reply]
My question is, where do you generate those logos? I want to have a custom one with my name on it.

**Reply by dotfrag** [reply]
> My question is, where do you generate those logos? I want to have a custom one with my name on it.

https://patorjk.com/software/taag/#p=display&f=ANSI%20Shadow&t=LazyVim

**Comment by nexusstar** [comment]
This is the updated version with snacks.nvim for anyone stumbling on this
```
return {
  "snacks.nvim",
  opts = {
    dashboard = {
      -- dashboard configuration
      preset = {
        header = [[ text ]],
      },
    },
  },
}
```

**Reply by hzionn** [reply]
thanks!

**Reply by he7yz** [reply]
thx mate, just what i needed!

**Reply by CarlosDanielDev** [reply]
It reaaly works fine:

```lua
{
    "folke/snacks.nvim",
    opts = {
      dashboard = {
        preset = {
          header = [[ text  ]],
        },
      },
    },
  },
```
Visite [here](https://patorjk.com/software/taag/#p=display&f=ANSI%20Shadow) and make your logo

**Reply by SnowFox4004** [reply]
This works. very nice
